FanNoiseSignal_Checker/Program.cs

```diff
@@ -0,0 +1,84 @@
+namespace FanNoiseSignal_Checker
+{
+    using Microsoft.Windows.EventTracing;
+    using System;
+    using System.IO;
+
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.Error.WriteLine("Usage: FanNoiseSignal_Checker.exe <AcpiFanNoiseImpact.etl>");
+                return;
+            }
+
+            if (!string.Equals(Path.GetExtension(args[0]), ".etl", StringComparison.OrdinalIgnoreCase))
+            {
+                Console.Error.WriteLine("Incorrect File! Please check the ETL file path again.");
+                return;
+            }
+
+            try
+            {
+                ProcessFanNoiseSignal(Path.GetFullPath(args[0]));
+            }
+            catch (Exception ex)
+            {
+                Console.Error.WriteLine($"An error occurred: {ex.Message}");
+            }
+
+            Console.WriteLine("Press any key to exit...");
+            _ = Console.ReadKey();
+        }
+
+        private static void ProcessFanNoiseSignal(string filePath)
+        {
+            var resultFilePath = Path.GetFullPath(@"FanNoiseSignal_Result"+ "_" + DateTime.Now.ToString("yyyyMMdd_HHmm") + ".txt");
+
+            using (var resultWriter = new StreamWriter(resultFilePath))
+            {
+                var trace = TraceProcessor.Create(filePath);
+                var pendingGenericEvent = trace.UseGenericEvents();
```

Comment:
Small optimization would be to add the provider ID for Microsoft.Windows.Kernel.Power into the "UseGenericEvents".